crates/common_utils/src/id_type/global_id.rs

```diff
@@ -79,6 +79,23 @@ impl CellId {
     }
 }
 
+impl<'de> serde::Deserialize<'de> for CellId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let deserialized_length_id = LengthId::deserialize(deserializer)?;
```

Comment:
can you deserialize to string, and then call `from_str` function?

Reply:
You can refer to the similar implementation here 
https://github.com/juspay/hyperswitch/blob/a31dcea30ded6360a0db952b828f188fd05a60d9/crates/common_utils/src/id_type.rs#L182